backend/api/pipeline.proto

```diff
@@ -320,6 +320,15 @@ message Pipeline {
   repeated ResourceReference resource_references = 9;
 }
 
+enum PipelineSpecFormat {
+  PIPELINE_SPEC_UNSPECIFIED = 0;
+  // In v1, we use argo workflow spec as the pipeline spec format.
+  V1 = 1;
+  // In v2, we temporarily use [api/v2alpha1/PipelineJob](https://github.com/kubeflow/pipelines/blob/19b3de4dd049052de74868ad0dfc931a4c700265/api/v2alpha1/pipeline_spec.proto#L11-L12)
+  // as the pipeline spec format.
+  V2_JOB = 2;
```

Comment:
NIT: Just a future looking consideration: If KFP is going to have v3.x.x or above, it will be a little strange to say that KFPv3 uses V2 format. Therefore, directly calling out the name of each format reduces confusion: Replace `V1` with `ARGO_WORKFLOW`, replace `V2_JOB` with `IR_PIPELINE_JOB`, and in the future, use `IR_PIPELINE_SPEC` for pipeline_spec format. How do you think?

Reply:
You raised some good arguments, I am now leaning towards naming it `V2ALPHA1_PIPELINE_JOB`.
At the time when we bump the version from v2alpha1 to v2, it's likely there are breaking changes, so they should be different spec formats.

Here's how I perceive it. The api is its own package and has its own version: https://github.com/kubeflow/pipelines/tree/master/api/v2alpha1.
Right now it's v2alpha1. It's okay for the api version to not match KFP version.
I think most likely if we ever want to build KFP v3, it will be the time when we want to bump the api version to v3, so they won't be out of sync.

Reply:
Sounds good! Then I think documenting the meaning of version here (paste the IR link to comments) will clear my confusion. Sounds good on your approach for combining proto_version+format_name!

Reply:
I thought the naming of enum here should be more related to the version of IR schema instead of the version of api.